File: backup_before_industrialisation_20250626/backend/metier_generator/templates/threed/api/middlewares/logger_middleware.py
"""
Middleware logger API threed (Python) - Ultra avancé
Journalisation structurée, corrélation, log métier, exportable
Utilisation : app.before_request(logger_middleware)
"""
import uuid
import time
from flask import g, request


def logger_middleware():
    if not hasattr(g, 'correlation_id'):
        g.correlation_id = request.headers.get('X-Correlation-Id', str(uuid.uuid4()))
    g.start_time = time.time()

    @request.after_this_request
    def log_response(response):
        duration = int((time.time() - g.start_time) * 1000)
        log = {
            'correlationId': g.correlation_id,
            'method': request.method,
            'url': request.path,
            'status': response.status_code,
            'duration': duration,
            'user': getattr(g, 'user_id', None),
            'businessContext': getattr(g, 'business_context', None)
        }
        print('[API-LOG]', log)
        return response

File: backup_before_industrialisation_20250626/backend/metier_generator/templates/threed/api/middlewares/logger_middleware.js
// Middleware logger API threed (Node.js) - Ultra avancé
// Journalisation structurée, corrélation, log métier, exportable
// Utilisation : app.use(require('./logger_middleware'));

const { v4: uuidv4 } = require('uuid');

function enrichLog(req, res, next) {
  req.correlationId = req.headers['x-correlation-id'] || uuidv4();
  req.startTime = Date.now();
  res.on('finish', () => {
    const duration = Date.now() - req.startTime;
    const log = {
      correlationId: req.correlationId,
      method: req.method,
      url: req.originalUrl,
      status: res.statusCode,
      duration,
      user: req.user ? req.user.id : null,
      businessContext: req.businessContext || null
    };
    console.log('[API-LOG]', JSON.stringify(log));
  });
  next();
}

module.exports = enrichLog;

// Exemple d'intégration avancée :
// const express = require('express');
// const app = express();
// app.use(require('./logger_middleware'));
// ...
